File: solutions/javascript/lasagna-master/1/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

/// <reference path="./global.d.ts" />
// @ts-check

export function cookingStatus(remainingTime) {
    switch (remainingTime) {
        case undefined:
            return 'You forgot to set the timer.'
        case 0:
            return 'Lasagna is done.';
        default:
            return 'Not done, please wait.';
    }
}

export function preparationTime(layers, avgPrepTime = 2) {
    return layers.length * avgPrepTime;
}

export function quantities(layers) {
    const NOODLE_PER_LAYER = 50;
    const SAUCE_PER_LAYER = 0.2;
    let result = {
        noodles: 0,
        sauce: 0
    }
    for (let layer of layers) {
        if (layer == 'noodles')
            result.noodles += NOODLE_PER_LAYER;
        else if (layer == 'sauce')
            result.sauce += SAUCE_PER_LAYER;
    }
    return result;
}

export function addSecretIngredient(friendsList, myList) {
    let secretItem = friendsList[friendsList.length - 1];
    myList.push(secretItem);
}

export function scaleRecipe(recipe, portions) {
    const factor = portions / 2;
    let result = {};
    for (let ingredient in recipe) {
        result[ingredient] = recipe[ingredient] * factor;
    }
    return result;
}

File: solutions/go/lasagna-master/1/lasagna_master.go
package lasagna

func PreparationTime(layers []string, avgPrepTimePerLayer int) int {
	if avgPrepTimePerLayer == 0 {
		avgPrepTimePerLayer = 2
	}
	return len(layers) * avgPrepTimePerLayer
}

func Quantities(layers []string) (int, float64) {
	countSauce := 0.0
	countNoodles := 0
	for _, layer := range layers {
		switch layer {
		case "sauce":
			countSauce += 0.2;
		case "noodles":
			countNoodles += 50;
		}
	}
	return countNoodles, countSauce
}

// TODO: define the 'AddSecretIngredient()' function
func AddSecretIngredient(friends []string, mine []string) {
	mine[len(mine)-1] = friends[len(friends)-1]
}

// TODO: define the 'ScaleRecipe()' function
func ScaleRecipe(quantities []float64, portions int) []float64{
	result := []float64{}
	for _, quantity := range quantities {
		result = append(result, quantity * float64(portions) / 2)
	}
	return result
}

// Your first steps could be to read through the tasks, and create
// these functions with their correct parameter lists and return types.
// The function body only needs to contain `panic("")`.
// 
// This will make the tests compile, but they will fail.
// You can then implement the function logic one by one and see
// an increasing number of tests passing as you implement more 
// functionality.
